tsdb/points.go

```diff
@@ -1309,25 +1308,39 @@ func (p Fields) MarshalBinary() []byte {
 		case int:
 			b = append(b, []byte(strconv.FormatInt(int64(t), 10))...)
 			b = append(b, 'i')
-		case int32:
+		case int8:
 			b = append(b, []byte(strconv.FormatInt(int64(t), 10))...)
 			b = append(b, 'i')
-		case uint64:
-			b = append(b, []byte(strconv.FormatUint(t, 10))...)
+		case int16:
+			b = append(b, []byte(strconv.FormatInt(int64(t), 10))...)
+			b = append(b, 'i')
+		case int32:
+			b = append(b, []byte(strconv.FormatInt(int64(t), 10))...)
 			b = append(b, 'i')
 		case int64:
 			b = append(b, []byte(strconv.FormatInt(t, 10))...)
 			b = append(b, 'i')
+		case uint:
+			b = append(b, []byte(strconv.FormatUint(uint64(t), 10))...)
```

Comment:
Could we check when we parse back if it starts with a `-`?  If it does, it's ParseInt, if not, it's ParseUint.  Would that work?